### ISSUE TITLE
UI flow enhancement ( after the board review and demo to a potential real user)

### ISSUE THREAD

**Issue by achille005** [achille005/JobApp#41]
- Remove email verification - Directly provide login Screen ( instead  of "Email verification screen - This email verification is useless because we saving the data before it is done)
-Move "First Name" "Last Name" to Log In Screen
- Remove Country, City and State from JS and R Profiles (only Job and Resume Profile have them) - Call it Job Location - make it such it look and feel like Country City and State are under "Job Location"
- keep the space between the field consistant

- Make the Skill fields same height like the other one because it does not take 2 lines anyway

- Move "What channel to post your profile to" to its own screen. 
- Remove Progress bar ( it takes too much space)
- "How often to receive email Alert" will be only available on Dashboard setting

![the new flow](https://cloud.githubusercontent.com/assets/6147947/26665397/551fc6b0-4667-11e7-973b-288d1abb303b.png)

**Comment by Astrea123** [comment]
For issue #41 , These are the changes required 

Recruiter flow 
1)For Selecting Category ,Now need to create new Screen that will store info of job Category with the validation
2) After selecting Control will move on job Profile Screen 
3)After that Recruiter will enter the Personal information  with Recaptcha

Changes Required for this flow
Extra thing One Screen  and recaptcha 
This flow will require new validations, way of storing info at client side and  new logic to send info in db on  server side  and need to write code of recaptcha.


JobSeeker Flow
1)For Selecting Category ,Now need to create new Screen that will store info of job Category with the validation
2)After Selecting the Category, JobSeeker can enter Resume Profile Detail.This page include new fields country,state,city,zipcode So info of these fields will save in Resume_Profile__c Object
Currently these fields are on JobSeeker Information page So info of these fields is in Contact object
3)After Entering info of Resume Profile ,User can  upload resume
4)After Uploading file, User can enter info about himself that will store the in “Contact ” object.Country,state,city,zipcode is not the contact object

Changes Required for this flow

This flow  has four step registration
New Screen for Job category and uploading resume with the Recaptcha functionality
This will require validations, way of storing info at client side and changes in the logic of  server side  and need to write code of recaptcha

Since country, state,city ,zipcode is  move to Resume_Profile__c object So I need to make changes in the Email cronjob code and JobSeeker dashboard also
in this issue you have asked for css changes also ,So I need to check that also
We need to create new test cases for that and also need to do testing again.


 It will take extra time of 2 or 3 Weeks . Please let us know if it fine  for you or you want to shift this in next release

Please Correct me if I misunderstood any thing

**Comment by achille005** [comment]
Please leave the backend alone and just restructure the UI, You have most UI code and validation already. 

you said:
 "country, state,city ,zipcode is move to Resume_Profile__c object So I need to make changes in the Email cronjob code and JobSeeker dashboard also"

Each UI screen does not have to match with an Object. No?
you can have "country, state,city ,zipcode " and "Job Tile, skills,..." on the same screen but save them in different Objects. No?
Conceptually, it does makes a lot of sens to save "country, state,city ,zipcode " in the "Resume Object" 

This changes have to be in this first release as we are not adding a new functionality

**Comment by achille005** [comment]
Also looking at these plans here (https://devcenter.heroku.com/articles/heroku-postgres-plans) - it is affordable. That is, we should be able to do the 3 resume upload types.
The argument around this being: The main reason why the Recruiter will want to pay for an app like this is find resumes and we are limitting to JS ways of adding resume to the app -- many people were not enthusiastic about it

**Comment by achille005** [comment]
What you should do is instantiate a resume  and JS/R profile object at the beginning of the flow 
and keep setting/populating it from screen to screen and on the last screen you save it into DB

**Comment by Astrea123** [comment]
I have couple of question about this flow 
1)Should I assign default value of "How often to receive email Alert" field or it will be empty ?
2)You want Attach resume as a separate screen  right ? ,here "Skip for now" options is for if user does not want to select the file but if user want to select the file there is no button to move on  after selecting the file

**Comment by achille005** [comment]
1)Should I assign default value of "How often to receive email Alert" field or it will be empty ?
>> yes. Default it to "Daily"
2)You want Attach resume as a separate screen right ? ,here "Skip for now" options is for if user does not want to select the file but if user want to select the file there is no button to move on after selecting the file
>> i missed "Continue" button on that screen and this "Continue" button has the same functionality as "Skip for now" button - If pressed show that pop with that message

**Comment by achille005** [comment]
@With all the algorithm issues on job title vs skill set, my team has decided to make a simple as possible so that we end up with an application that works. Here are the changes my team has decided:

- Remove Logo from Home Page ( Show the logo for 1 second when app is loading for the first time). Also use it as the app icon and the browser tab icon. Note: Use Logo on Web Version.
- Remove Skills Set field from Front end and backend.
- On Home Page make  "Last name" and "first name" as hidden field but hard-code them as  "First Name" and "Last Name" values to satisfy salesforce required fields - In Settings tab on dashboard leave these 2 fields blank so that the users can update them on dashboard if they choose to.
- Remove Category Screen from front-end and backend and from the code.
-  From Home Page when "Start Receiving New Job Alerts!" See new design below button is pressed, actually **Save the data into the database**
- Note:
Home page is now made of 2 tabs ("Job Seekers" and "Employers")  . "Job Seekers" is the default should be more bold and bigger font and 
- Now you can use the match() function to match at least 1 word to make it an acceptable match

![ui flow 2](https://user-images.githubusercontent.com/6147947/27014710-f50a610a-4ecb-11e7-90c9-1210cb1b9307.png)

**Comment by achille005** [comment]
- Key: when the user press "Start Receiving new Candidate Alerts" or "Save & Add Resume" the data is saved in db.


![ui flow 3](https://user-images.githubusercontent.com/6147947/27260534-7f2731e4-53fc-11e7-8e5a-33aee1518e19.png)

**Comment by achille005** [comment]
- 1
>**Hide Firstname and Lastname : Name of person is his/her mode of identification. If JobSeeker simply post his profile and do not login on dashboard and change his name, then how will recruiter identify the person he is giving job to.** 

 _-Since this is a mobile app, we want to collect the most important information from the user as soon as possible so that  our cron jobs can start working and reach out to them. That is why we are saving the the data to salesforce backend on second screen when "Start Receiving new Job/Candidate Alerts" button is pressed. Now, to what Payal told me, first name and last name are required fields to save to contact object on salesforce. If that information is not correct then we don't need those 2 fields as place holders till we get to login screen (last screen).  If we don't have the name of the JS, Recruiter (R ) can still email him/her, ask for resume, that is all all we care about. All this app is doing is to connect JSs and Rs._
**>Also this point contradict to screenshot send by you it contains firstname and lastname fields.Also it is important when job is match with the jobseeker then notification will go to the jobseeker which require firstname otherwise it will show "Hi first name" in the mail**
_No contradiction: if the user get to login screen( last screen) they will update "First Name" and "Last Name" used earlier as place holders -  
If JS name is not known yet( or check if it is "First Name" just say "Hi," in the email notification and don't mention it on Recruiter Dashbaord._
- 2
**>You are showing some input fields in label which is confusing like company name and email address they should be in input text.**
_Those are not input fields. they are UNeditable fields. After you save email address and company name on second screen as described in 1.), on the next screen you can display them back to the user(just to have them review what they have saved.)_

- 3
**>Email Address on Resume Uploaded: Email address is on the next page then how can we show email info on resume page.**
_Same as above in 2.) Once you save the info on screen, you have it in the session so you can display it on the subsequent screens._


- 4
**Login with Social Sites was on login Screen ,you have give header name as login to the registration screen.I have checked dice.com they don't have social login option on registration page ,they have this functionality on login screen.**
_If you see the note below the login screenschot ( No. 4 ) i said, monster.com and indeed.com
Obviously we want to use the same social medias during login and account creation process. I think facebook and linkedIn are enough._

- 5 
**>You have shown social sharing on every page what should I share on those social sharing.**

_As the title said: "Tell a friend about JobApp" sharing the app LOGO as the home page link with hard coded title/subtitle "Looking for a new Job? Try JobApp" in Bold font.  The current functionality on home page is correct. Just copy and paste on the footer of the other screens_ 

- 6
**>The UI you have send it will be look nice for mobile but for desktop it is not looking so promising** 
_Maybe we can touch the desktop one a little, but we care more about how it looks on Mobile and easy it is to use ( make the user type as less as possible)_

**Comment by achille005** [comment]
- Few Notes
========
-instead of "go relax, i'll find you one" say "go relax, JobApp will find you one"
-"JobApp will never post content to your personal social pages." added to the screen shot
-Note Recruiter does not have "Upload Resume" screen
-Note the button on second screen for JobSeeker flow is "Save and Add Resume" now

**Comment by Astrea123** [comment]
I have noticed that there is no "preference field" in the image ,Do you want to remove it?
Please reply asap.

**Comment by achille005** [comment]
It is only available on Dashboard.

**Comment by Astrea123** [comment]
Social login functionality is different on monster.com and indeed.com ,
**Monster**
when we click on social login button it take info from Social login site and fill the form 
**Indeed** 
when we click on Social login button then it will create account in database  

Which one you prefer?

**Comment by achille005** [comment]
The best one is Monster because it gives the user the chance to change their detail before saving it to db )

**Comment by achille005** [comment]
I see you deployed the new design, 
- please try to make it fit on 1 screen so that the user does not have to scroll ( that is our goal, no scrolling, and less typing)
   . For example, too much space between the " Residence fields
   - social Media logo are too big - Please " Tell a friend about JobApp" lower case
- Remove the box around recaptcha ( i'm not a robot). It is darker, different from the other boxes (e.g place of residence box is lighter)
- Rmove "Resume Profile" Title
-In general make the space between the field proportional ( e.g: https://www.dice.com/register)
- Let us not make the password choice complicated - remove the 8 char minimum validation.
- Provide "Keep me signed in" by default so that the new registered can stay signed in
- After the last screen ( first name, lastname, email, password screen) The user should be directly take to his dashboard -- Not to the login screen
- The whole orange button on home page should be clickable (like before) and take the user to next screen ( not just the "Next >>" 
- mistyped( cap/cases and "will" was missed) - Just copy and paste this below 
  Looking for a new job? 
SETUP A NEW JOB ALERT,  and go relax, JobApp will find you one...

**Comment by achille005** [comment]
After a lengthy discussion we have decided to include "Your Job Category" screen right after the second screen. ( see the illustration below.... 

![new ui flow](https://user-images.githubusercontent.com/6147947/27522327-36ec22f2-59f3-11e7-9295-a4e10a9c8f60.png)

- We can talk about it but i think the login screen above is confusing because if the user choose to do social media login, he won't need to press "Log In to view new Job Openings" - see monster.com.
 ( Change this button to read "Log In to view new Job Openings")  
The new log in screen should look like:
![image](https://user-images.githubusercontent.com/6147947/27526446-e70319f0-5a13-11e7-87a3-1659c9592587.png)

**Comment by Astrea123** [comment]
I have checked monster site . In India Social login functionality is different . That's why I have asked you 

" Social login functionality is different on monster.com and indeed.com ,
**Monster**
when we click on social login button it take info from Social login site and fill the form
**Indeed**
when we click on Social login button then it will create account in database
Which one you prefer?"


**After that you replied** 
"The best one is Monster because it gives the user the chance to change their detail before saving it to db )"

I have prepared a video of this Please take a look.
https://drive.google.com/file/d/0B5yTvhVBvdrqZ190VU9BU0JrbDA/view?ts=5952405e

I have checked
 https://login20.monster.co.uk/Become-Member/Create-Account?intcid=swoop_TopNav_Register 

It has different logic if you want  functionality like this page has ,then in my opinion Social login functionality should apply on first screen because we have already filled the email address and user will be confuse after that.

**NOTE : We cannot fetch password from the Social login**

**Comment by achille005** [comment]
- In the video, you are using g+ - I know before we have decided to use g+ f and in but if we are using only f and in for signup process the log in also should just use those.
- Try to log out from facebook on your computer ( no existing session ) and try to "continue with facebook" and see .  It shouldn't fetch the user detail with option to edit them then automatically authenticate the user with their social media account and take user to his jobapp dashboard - **The user won't need to enter password** ( that is our goal no typing).

- If the user has already enter an email address and decide to use social media account then social media one will automatically overwrite the existing one otherwise the authentication will fail

- Use a bran new fb and linked In account to do this.   Let me know.

**Comment by achille005** [comment]
You got my point above? It is important for the app.

**Comment by achille005** [comment]
Also putting this link here since it is UI related:
https://github.com/userfor544/JobApp/issues/53

**Comment by Astrea123** [comment]
Yes , I got your point. We have put option to do social media login on profile screen of jobseeker and recuiter. if JobSeeker create account using social media then resume and contact info will save in db. if Recruiter create account using social media then job and contact info will save in db and user will move on dashboard. For login from social media we have provided button  on login page from there he can login simply.

New account will be create during registration time only. When user use social login on login page no new account will be create, only this will be checked if user have already registered then he will move to dashboard otherwise alert will be shown.

**Comment by achille005** [comment]
- Do you agree that if the user choose to login with social media(facebook and linkedIn account, there will be no need to enter password? 

 - Do you see why this screen makes more sens than the previous one:
Use Case with this screen should be: Choose one of the social media button > if existing there is an existing social media login session in your browser, fetch first name, last name for user to edit them if needed. > Continue button pressed > user is authenticated remotely with social media account > and directed to app dasboard. 

> if you have no soical media session, user is asked to create a social media account on social media site > then fetch fName and lName for edit > then authenticate > dashabord 
![image](https://user-images.githubusercontent.com/6147947/27526446-e70319f0-5a13-11e7-87a3-1659c9592587.png)

**Comment by Astrea123** [comment]
Yes I agree that if user choose to login with social media then no password is required.

1) Right now what we are doing is if user click social media button then 
   --his new account will be created. 
   --Firstname and Lastname are fetched from social media account and same will be stored. 
   --Mail id of social media account is stored in database. 
   --For password we are not storing anything in database. 
   --During login process from social media same password as of social media account will be used for authentication. Social media will authenticate the user.

2) If user create his account by normal process then
	-- he will fill value of firstname, lastname, password and then his account will be created.
	-- as during login user can't login only by using username only, password is also must.
	--So we have make password as required field (for doing login password is necessary) We can't remove validation from password field.
	

3) If we do as mentioned in latest github issue 41 (During Registration)
	--When User will click on social media button Information of user will fetch from his social media account 
	--We will insert firstname, lastname, email on login page and then allow user to edit firstname and lastname.
	--You don't want user to enter password and we can't fetch password from social media, they don't allow third party app to use users password, as this will breach privacy of user.
	-- Since we have applied validation on password field to "match the password" and "password as a required field" then we will remove those validation if user has choosen the social login.

**Comment by achille005** [comment]
Since we have applied validation on password field to "match the password" and "password as a required field" then we will remove those validation if user has choosen the social login.
- Use another UI ( or hide or grayed out  password fields) so that there is no need to worry about validation.  
        . on this screen user can edit fname and lname and press "Log In & ... " button. 
                . If there is a log on session(user has already logged in on facebook on that computer) the user is authenticated automatically and directed to dashboard
                . If there is no log on session user is taken to socia media login screen.
e.g: As i advised you, try to close log out of facebook close all the browser and try the flow and see what happened.

**Comment by Astrea123** [comment]
We had almost made the changes you suggested. We will deploy the package soon after testing it.

**Comment by achille005** [comment]
- The log in screen during the registration was  a little confusing to me.. I would suggest to completely hide the password fields and recaptcha ( if you still have it)
- minor touch on Category Screen. do like below:
- i tried to point my finger on the i ( next to Tech Jobs) on my mobile phone (it works on my PC) but no popup came out... 
![image](https://user-images.githubusercontent.com/6147947/28052277-0a347a16-65d8-11e7-8d0d-5a2d757ff34c.png)
